resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/kubernetes/integrationtest/KubernetesSuite.scala

```diff
@@ -364,8 +308,12 @@ private[spark] class KubernetesSuite extends SparkFunSuite with BeforeAndAfter {
         .pods
         .withLabel("spark-app-name", "spark-file-existence-test")
         .watch(watch)) { _ =>
-      SparkSubmit.main(args)
```

Comment:
`SparkSubmit.main` is unwieldy because

(1) It's difficult to re-use arguments since they have to be formatted as an array rather than as a Map, and more importantly
(2) They deal with system properties which is global state across all tests in the JVM.